NorimaConsulting/training-and-onboarding
# REC-35: Set up React GraphQL library and Codegen

- Set up React GraphQL library along with Codegen and installed the proper dependencies (followed this [document](https://github.com/tesseractcollective/react-graphql-docs) as provided in Jira Ticket)
- Included 'UsersQuery.tsx' that includes a test GraphQL query whose result is shown on '/testQuery' route

Testing:

1. Add .env file in root directory with the following variables (see `.env.sample` file for reference):
- `VITE_HASURA_GRAPHQL_ADMIN_SECRET`
- `VITE_HASURA_GRAPHQL_ENDPOINT="https://winter-2022.hasura.app"`
2. If needed, update the GraphQL query at variable `const getUsersQuery` in file `UsersQuery.tsx`. Currently it is set to retrieve all registered users from the database
3. Go to `/webapp` directory in terminal and run `pnpm dev`
4. Visit `http//localhost:3000/testQuery` and you should see a list of all users retrieved using the query
